src/connect/components/drive-view/drive-view.tsx

```diff
@@ -1,189 +1,82 @@
-    onCancelInput?: ConnectTreeViewProps['onCancelInput'];
-    onItemOptionsClick?: ConnectTreeViewProps['onItemOptionsClick'];
-    onDropActivate?: ConnectTreeViewProps['onDropActivate'];
-    onDragStart?: ConnectTreeViewProps['onDragStart'];
-    onDragEnd?: ConnectTreeViewProps['onDragEnd'];
-    onCreateDrive?: (drive: AddDriveInput | AddPublicDriveInput) => void;
-    disableAddDrives?: boolean;
-    displaySyncFolderIcons?: boolean;
-}
-
-const filterDriveByType = (drive: DriveTreeItem, type: DriveType) => {
-    switch (type) {
-        case 'PUBLIC_DRIVE':
-            return drive.type === 'PUBLIC_DRIVE';
-        case 'LOCAL_DRIVE':
-            return drive.type === 'LOCAL_DRIVE';
-        case 'CLOUD_DRIVE':
-            return drive.type === 'CLOUD_DRIVE';
-        default:
-            return false;
-    }
-};
+export type DriveViewProps = TUiNodesContext &
+    NodeProps &
+    DragAndDropProps & {
+        label: ReactNode;
+        groupSharingType: SharingType;
+        disableAddDrives: boolean;
+        className?: string;
+        showAddDriveModal: (groupSharingType: SharingType) => void;
+        showDriveSettingsModal: (uiDriveNode: UiDriveNode) => void;
+    };
 
 export function DriveView(props: DriveViewProps) {
     const {
+        driveNodes,
+        selectedDriveNode,
+        label,
+        groupSharingType,
         className,
-        type,
-        name,
-        onDropEvent,
-        onItemClick,
-        onSubmitInput,
-        onItemOptionsClick,
-        defaultItemOptions,
-        onDropActivate,
-        onDragStart,
-        onDragEnd,
-        onCancelInput,
-        disableHighlightStyles,
-        drivePath = '/',
-        onCreateDrive,
         disableAddDrives,
-        isAllowedToCreateDocuments = true,
-        displaySyncFolderIcons = false,
-        ...restProps
+        isAllowedToCreateDocuments,
+        showAddDriveModal,
     } = props;
-    const [showAddModal, setShowAddModal] = useState(false);
-
-    const drives = usePathContent(drivePath) as DriveTreeItem[];
-
-    const allowedTypes: TreeItemType[] = [
-        CLOUD_DRIVE,
-        FOLDER,
-        LOCAL_DRIVE,
-        PUBLIC_DRIVE,
-    ];
-    const allowedDrives = drives
-        .filter(drive => filterDriveByType(drive, type))
-        .map(drive => drive.path);
+    const hasDriveNodes = driveNodes.length > 0;
+    const isContainerHighlighted =
+        selectedDriveNode?.sharingType === groupSharingType;
 
-    const isPublicDrive = type === PUBLIC_DRIVE;
-    const isCloudDrive = type === CLOUD_DRIVE;
-    const isLocalDrive = type === LOCAL_DRIVE;
+    function onShowAddDriveModal() {
+        showAddDriveModal(groupSharingType);
+    }
 
     return (
         <div
             className={twMerge(
                 'border-y border-gray-100 pl-4 pr-1 first-of-type:border-b-0 last-of-type:border-t-0',
-                isPublicDrive && 'bg-gray-100 ',
+                hasDriveNodes && 'pb-2',
```

Comment:
After discussion with design, it was decided that we would rather do the gray background highlight for a group of drives for the group that has the selected drive in it. Previously it was always PUBLIC that was highlighted.